core/templates/components/button-directives/create-activity-button.component.spec.ts

```diff
@@ -0,0 +1,285 @@
+      // suppress this error because we need only "isLoggedIn" and
+      // "canCreateCollections" functions for testing.
+      // @ts-expect-error
+      .and.returnValue(Promise.resolve({
+        isLoggedIn: () => true,
+        canCreateCollections: () => true
+      }));
+    const urlParamsSpy = spyOn(urlService, 'getUrlParams').and.returnValue({
+      mode: 'create'
+    });
+    spyOn(component, 'initCreationProcess');
+
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(false);
+
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+
+    expect(userServiceSpy).toHaveBeenCalled();
+    expect(urlParamsSpy).toHaveBeenCalled();
+    expect(component.initCreationProcess).toHaveBeenCalled();
+    expect(component.canCreateCollections).toBe(true);
+    expect(component.userIsLoggedIn).toBe(true);
+  }));
+
+  it('should initialize and call createNewExploration()', fakeAsync(() => {
+    const userServiceSpy = spyOn(userService, 'getUserInfoAsync')
+      // This throws "Argument of type 'Promise<{ isLoggedIn: () => boolean;
+      // canCreateCollections: () => boolean; }>' is not assignable to parameter
+      // of type 'Promise<UserInfo>'.". This is because the actual
+      // 'getUserInfoAsync' returns more properties than required. We need to
+      // suppress this error because we need only "isLoggedIn" and
+      // "canCreateCollections" functions for testing.
+      // @ts-expect-error
+      .and.returnValue(Promise.resolve({
+        isLoggedIn: () => true,
+        canCreateCollections: () => false
+      }));
+    const urlParamsSpy = spyOn(urlService, 'getUrlParams').and.returnValue({
+      mode: 'create'
+    });
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(false);
+
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+
+    expect(userServiceSpy).toHaveBeenCalled();
+    expect(urlParamsSpy).toHaveBeenCalled();
+    expect(explorationCreationServiceSpy).toHaveBeenCalled();
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(true);
+  }));
+
+  it('should return from initCreateProcess() when' +
+    ' creation is in progress', () => {
+    component.creationInProgress = true;
+    expect(component.initCreationProcess()).toBe();
+  });
+
+  it('should create new exploration if user cannot create collections', () => {
+    component.creationInProgress = false;
+    component.canCreateCollections = false;
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    component.initCreationProcess();
+
+    expect(explorationCreationServiceSpy).toHaveBeenCalled();
+  });
+
+  it('should replace location', () => {
```

Comment:
Names should be of the form 

should do something if/when some "trigger" happens

Reply:
Renamed